src/core/server/csp/csp_directives.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CspConfigType } from './config';
+
+export type CspDirectiveName =
+  | 'script-src'
+  | 'worker-src'
+  | 'style-src'
+  | 'frame-ancestors'
+  | 'connect-src'
+  | 'default-src'
+  | 'font-src'
+  | 'frame-src'
+  | 'img-src'
+  | 'report-uri'
+  | 'report-to';
+
+/**
+ * The default rules that are always applied
+ */
+export const defaultRules: Partial<Record<CspDirectiveName, string[]>> = {
+  'script-src': [`'unsafe-eval'`, `'self'`],
+  'worker-src': [`blob:`, `'self'`],
+  'style-src': [`'unsafe-inline'`, `'self'`],
+};
+
+/**
+ * Per-directive rules that will be added when the configuration contains at least one value
+ * Main purpose is to add `self` value to some directives when the configuration specifies other values
+ */
+export const additionalRules: Partial<Record<CspDirectiveName, string[]>> = {
+  'connect-src': [`'self'`],
+  'default-src': [`'self'`],
+  'font-src': [`'self'`],
+  'img-src': [`'self'`],
+};
```

Comment:
We can't add new default rules for these directive, as it will be a breaking change, however we want to add `self` when the customer configure another source, to still allow to fetch from our server.

E.g if the customer configures `csp.img_src: ['foo.com']`, we'll be effectively using `img-src: 'self' foo.com`